### ISSUE TITLE
Using anitopy library for parsing names

### ISSUE THREAD

**Issue by ghost** [hotsno/mpv-anilist#6]
I think [Anitopy](https://github.com/igorcmoura/anitopy) is unmatched when it comes to parsing names, and it would be useful for this project I think, and its easy to use.

**Comment by hotsno** [comment]
Thanks for sharing!

But I've actually known about this even before I started working on Keroro. This unfortunately doesn't do the (in my opinion) hardest part of mapping file names to entries on AniList, which is finding the correct AniList entry given the episode number, season #, and anime name _as it appears in the file name_. The last part of that sentence is the important part.

Using the information as it appears in the file name and then querying AniList for the top result with that is going to be inaccurate _a lot_ of the time. I've looked at how some other projects handle this, and it's not something I want to invest my time and energy into. I could maybe use someone else's code as reference, but the one that I looked at was in JavaScript and involves so much potentially flaky logic that I wouldn't feel good using it without writing unit tests because I'd probably make several mistakes while rewriting it. This would add even more time and effort, given that it actually requires me to understand what all of this pretty tedious logic is doing.

Unfortunately, I don't think I'll ever be automating file name -> AniList entry mapping with Keroro, even though I would have liked for it to have that feature.

I've personally actually found manually mapping anime names to not be as annoying as I thought I would, and this is coming from someone who loves automation. It also guarantees it's always accurate.

---

If by chance anyone ever wants to work on this feature, feel free to leave a comment here beforehand, but most I'll almost certainly be open to merging a PR for this!
